### ISSUE TITLE
Predefined `Schema` for `Int` does not have `.format("int32")` by default

### ISSUE THREAD

**Issue by brainsap** [softwaremill/tapir#2027]
Looks like, according to [OAS](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#format), the default value for `format` attribute for `Int` should be `int32`.

Currently the schemas specified by Tapir in `Schema.scala` look like this:
```scala
  implicit val schemaForInt: Schema[Int] = Schema(SInteger())
  implicit val schemaForLong: Schema[Long] = Schema(SInteger[Long]()).format("int64")
  implicit val schemaForFloat: Schema[Float] = Schema(SNumber[Float]()).format("float")
  implicit val schemaForDouble: Schema[Double] = Schema(SNumber[Double]()).format("double")
```

Is there any reason for `Int` not having a predefined "int32" format by default? If not, it'd be great to have this.
Thanks!

**Comment by adamw** [comment]
Hm that's an interesting question. From a JSON perspective, shouldn't we have the following mappings:

* scala/java `Int` -> schema `integer` + format `int32`
* `Long` -> `String` + `int64` (since json number go up to 2^53, right?)
* `Float` -> `Number` + `float`
* `Double` -> `Number` + `double`

Though in theory, we aren't restricted to JSON schemas. These are also used for query/header values, though I think it doesn't really matter for those if the schema is an int or string, as long as the format is correct. Could be a different case for other formats, but then you can easily override the default schemas by providing your own implicits, which will always have a higher priority.

Summing up, I think we should change the schemas both for `Int` and `Long`, what do you think?

**Comment by ghik** [comment]
> (since json number go up to 2^53, right?)

JSON numbers are unlimited. You can represent every `Long` value just fine with a JSON number - as long as you don't parse it inside a JS runtime. Usage of REST/JSON does not imply JS in any way though.

**Comment by adamw** [comment]
@ghik makes sense :) so we'd only change the schema for `Int` to `int32` format
